qa/evil-tests/src/test/java/org/elasticsearch/common/logging/EvilLoggerTests.java

```diff
@@ -340,26 +340,22 @@ public void testPrefixLoggerMarkersCanBeCollected() throws IOException, UserExce
     }
 
     public void testProperties() throws IOException, UserException {
-        final Settings.Builder builder = Settings.builder().put("cluster.name", randomAlphaOfLength(16));
-        if (randomBoolean()) {
-            builder.put("node.name", randomAlphaOfLength(16));
-        }
-        final Settings settings = builder.build();
+        final Settings settings = Settings.builder()
+                .put("cluster.name", randomAlphaOfLength(16))
+                .put("node.name", randomAlphaOfLength(16))
+                .build();
         setupLogging("minimal", settings);
 
         assertNotNull(System.getProperty("es.logs.base_path"));
 
         assertThat(System.getProperty("es.logs.cluster_name"), equalTo(ClusterName.CLUSTER_NAME_SETTING.get(settings).value()));
-        if (Node.NODE_NAME_SETTING.exists(settings)) {
-            assertThat(System.getProperty("es.logs.node_name"), equalTo(Node.NODE_NAME_SETTING.get(settings)));
-        } else {
-            assertNull(System.getProperty("es.logs.node_name"));
-        }
+        assertThat(System.getProperty("es.logs.node_name"), equalTo(Node.NODE_NAME_SETTING.get(settings)));
     }
 
     public void testNoNodeNameInPatternWarning() throws IOException, UserException {
+        String nodeName = randomAlphaOfLength(16);
+        LogConfigurator.setNodeName(nodeName);
```

Comment:
Since we automatically add a node name to the pattern we need to configure one or else the NodeNamePatternConverter will refuse to be built.